File: backend/scripts/wrap_agencies.py
import os
import re

base_dir = os.path.dirname(__file__)
target_path = os.path.normpath(os.path.join(base_dir, '../../frontend/agencies.html'))

# Read the file
with open(target_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Find all agency cards that are NOT already wrapped
# Pattern to match agency cards not in a wrapper
pattern = r'(?<!<div class="agency-wrapper"[^>]*>\n)(<div class="agency-info compact">.*?</div>\n)'

# Counter for unique IDs
agency_count = 0

def wrap_agency(match):
    global agency_count
    agency_count += 1

    # Extract agency name from the card
    agency_card = match.group(1)
    name_match = re.search(r'<h3[^>]*>(.*?)</h3>', agency_card)

    # Create a simple ID based on count
    agency_id = f'agency-{agency_count}'

    # Create the wrapper with reviews box
    wrapped = f'''<div class="agency-wrapper" id="wrapper-{agency_id}">
{agency_card}
    <div class="past-reviews-box" id="reviews-{agency_id}">
        <h3>Past Reviews</h3>
        <div class="no-reviews-message">
            No reviews as yet
        </div>
    </div>
</div>

'''
    return wrapped

# This is complex because agency cards span multiple lines
# Let's use a different approach - find each card individually

# Split content by agency cards
lines = content.split('\n')
output_lines = []
i = 0

while i < len(lines):
    line = lines[i]

    # Check if this is a standalone agency card (not already in wrapper)
    if '<div class="agency-info compact">' in line:
        # Check if previous line is NOT a wrapper opening
        if i == 0 or 'agency-wrapper' not in lines[i-1]:
            # Found a standalone card, need to wrap it
            agency_count += 1
            agency_id = f'agency-{agency_count}'

            # Add wrapper opening
            output_lines.append(f'<div class="agency-wrapper" id="wrapper-{agency_id}">')
            output_lines.append(line)

            # Continue until we find the closing </div> for this card
            i += 1
            depth = 1
            while i < len(lines) and depth > 0:
                line = lines[i]
                if '<div' in line and 'class=' in line:
                    depth += 1
                if '</div>' in line:
                    depth -= 1

                output_lines.append(line)

                if depth == 0:
                    # Add reviews box
                    output_lines.append('')
                    output_lines.append(f'    <div class="past-reviews-box" id="reviews-{agency_id}">')
                    output_lines.append('        <h3>Past Reviews</h3>')
                    output_lines.append('        <div class="no-reviews-message">')
                    output_lines.append('            No reviews as yet')
                    output_lines.append('        </div>')
                    output_lines.append('    </div>')
                    output_lines.append('</div>')
                    break

                i += 1
        else:
            # Already wrapped, just copy
            output_lines.append(line)
    else:
        output_lines.append(line)

    i += 1

# Write the output
output_path = os.path.normpath(os.path.join(base_dir, '../../frontend/agencies_wrapped.html'))
with open(output_path, 'w', encoding='utf-8') as f:
    f.write('\n'.join(output_lines))

print(f"Wrapped {agency_count} agency cards")

File: backend/scripts/wrap_agencies.js
const fs = require('fs');
const path = require('path');

const targetPath = path.resolve(__dirname, '../../frontend/agencies.html');

// Read the file
let content = fs.readFileSync(targetPath, 'utf8');

// List of all agency IDs
const agencyIds = [
    '10881', 'A1', 'Access', 'Acep', 'Adnil', 'Akey1', 'Alphanso', 'Arize',
    'Atlantic', 'Atsed', 'Beadles', 'CapeCod', 'CCS', 'Checkin', 'Csa', 'Cuse',
    'CY', 'Dallas', 'Destin', 'Dgs', 'DJ', 'Dorryus', 'Ejam', 'ExploreWork',
    'ExploreYourWorld', 'FaithPlacement', 'FirstClass', 'FlyerZone', 'Gamma',
    'GlobalInsight', 'GoldenLuxe', 'GTG', 'HRElite', 'ICDE', 'InterExchange',
    'IRSS', 'IslandPlus', 'ITCE', 'IWTS', 'JAB', 'JD', 'Joyst', 'Laconic',
    'LenClair', 'Lils', 'MaxJ', 'Mayfos', 'Mcleans', 'MH', 'Owt', 'Passport',
    'Patriots', 'Platinum', 'Polaris', 'Practical', 'Progressive', 'Seaview',
    'Seramil', 'Skills', 'Snow', 'Sora', 'Steep', 'Summerchoice', 'Swat',
    'Tander', 'Thelia', 'Trailes', 'Travelaire', 'TrevorHamilton', 'WorkAbroad'
];

// Already wrapped IDs (from grep count)
const wrapped = ['10881', 'Arize', 'A1', 'Access', 'Adnil', 'Acep', 'Akey1', 'Alphanso', 'Atlantic', 'Atsed', 'Beadles'];

// IDs that need wrapping
const needsWrapping = agencyIds.filter(id => !wrapped.includes(id));

console.log(`Need to wrap ${needsWrapping.length} agencies`);

// Process each agency
let wrappedCount = 0;
for (const agencyId of needsWrapping) {
    // Pattern to find the submit button for this agency
    const submitPattern = `onclick="submitReview${agencyId}()">Submit Review</button>`;

    if (!content.includes(submitPattern)) {
        console.log(`Skipping ${agencyId} - not found`);
        continue;
    }

    // Find the pattern: submit button, closing tags, leave review button, closing div, then next card
    const pattern = new RegExp(
        `(submitReview${agencyId}\\(\\)">Submit Review</button>[\\s\\S]*?` +
        `<button type="button" class="btn btn-primary" onclick="toggleReviewSection\\(this\\)">Leave a Review</button>\\s*` +
        `</div>)\\s*\\n\\s*\\n\\s*` +
        `(<div class="agency-info compact">)`,
        'g'
    );

    const replacement = `$1\n\n    <div class="past-reviews-box" id="reviews-${agencyId.toLowerCase()}">\n` +
        `        <h3>Past Reviews</h3>\n` +
        `        <div class="no-reviews-message">\n` +
        `            No reviews as yet\n` +
        `        </div>\n` +
        `    </div>\n` +
        `</div>\n\n` +
        `<div class="agency-wrapper" id="wrapper-${agencyId.toLowerCase()}">\n    $2`;

    const newContent = content.replace(pattern, replacement);

    if (newContent !== content) {
        content = newContent;
        wrappedCount++;
        console.log(`Wrapped ${agencyId}`);
    } else {
        console.log(`Could not wrap ${agencyId}`);
    }
}

// Handle the last agency - it won't have a next card to match
const lastAgency = needsWrapping[needsWrapping.length - 1];
const lastPattern = new RegExp(
    `(submitReview${lastAgency}\\(\\)">Submit Review</button>[\\s\\S]*?` +
    `<button type="button" class="btn btn-primary" onclick="toggleReviewSection\\(this\\)">Leave a Review</button>\\s*` +
    `</div>)\\s*\\n`,
    'g'
);

const lastReplacement = `$1\n\n    <div class="past-reviews-box" id="reviews-${lastAgency.toLowerCase()}">\n` +
    `        <h3>Past Reviews</h3>\n` +
    `        <div class="no-reviews-message">\n` +
    `            No reviews as yet\n` +
    `        </div>\n` +
    `    </div>\n` +
    `</div>\n`;

content = content.replace(lastPattern, lastReplacement);

// Write the result
fs.writeFileSync(targetPath, content, 'utf8');

console.log(`\\nTotal wrapped: ${wrappedCount} agencies`);
console.log('File updated successfully!');
